### PR TITLE
RFC 0014 -- Address Encoding

### PR DESCRIPTION
**Pull request by o1brad**
Added RFC encouraging us to change to a Base58 encoding. 

Co-authored-by: Brandon Kase <bkase@users.noreply.github.com>

### CODE REVIEW

**Comment by nholland94** [comment]
Sounds good to me. We should put a `to_string_hum` or `to_base58_string` to the `Data_hash` functors. We should also have custom `yojson` and `sexp` serializers/deserializers on hashes so that we get this conversion for free most places we log hashes.

**Comment by cmr** [comment]
We're probably going to end up baking this decision into the protocol forever. Seems fine to me. I like that it includes a basic error-detecting code.

File: rfcs/0013-address-encoding.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+## Summary
+[summary]: #summary
+
+Change address encoding from Base64 to Base58.
+
+## Motivation
+
+[motivation]: #motivation
+
+First, it's the standard format across other well known cryptocurrencies (Ethereum, Bitcoin). That means most entities in the ecosystem, from end-users to exchanges, will be more familiar and comfortable with Base58 encodings than what we have currently.
```
Comment (enolan) [inline rfcs/0013-address-encoding.md]: Ethereum uses hexadecimal with a cute parity scheme based on capitalizing some letters.

**Comment by enolan** [comment]
Actually it'd be best if Coda addresses didn't even *look* like Bitcoin addresses. Not sure that's something we can easily accomplish while sticking to the Base58 encoding though. Maybe a prefix? Like all Coda addresses start with CO or something.

**Comment by jkrauska** [comment]
Why do all current keys start with K and end with Bs and As?

```
 1, KImusi2ZkuWGPXN0YoOiOYRwIVLTqj91h1lYCbosEBDza7AEtmgBAAAB
 2, KF/gCh9fJ7rsTA72qygPI+BW1hLOGmNHIj+jXsv30WJpIkrVxzsBAAAA
 3, KPmGh0Hcd+reZhLFrPOhWcYMkSi4qvfkGNkO2g0i1Sz9M8IPakABAAAA
 4, KF5Lzhu3HEKR20EaK8bwKBbZKKVBRRowuYOLFIqye5MgMi4DqLADAAAA
 5, KPwcHBWoR9QzBsm/g46deD7eGFTId0n1xxBV4VoCcDBB3ik4/ekCAAAA
 6, KF4DPWVIdbXOLus9wM/HZS0rIFnX+YpGO9jnAdFt7xLR2QNB66EBAAAB
 7, KEYVC4/00VIjid9Sk7xGzZ46dQ7mWlZ5eFeSSnD7OLvw+NKq25IAAAAA
 8, KKtOMsn5QagJty8uN/9uJeA18V/4ZHWAZl2tqR391DcOBCSiPlYBAAAB
 9, KB8X1InBWdbUBnknkjz33c2MaCvzxSZqqJugHwK5cCQysb87v+AAAAAB
10, KFhYqUSIKJHkk8ELghLu27TMwUw4zEwwdIwVn06y/z5BebexrPgBAAAA
11, KHYImaHvbNz3wCv5w2TpJLrZmLKP5y6YcYk6grOzpdXhF4cFcxIDAAAA
12, KGmEwVccJY/A9dYFzR16F5bmSS7r3HgArxhVLhpmyhClWeEPWoACAAAB
13, KKjyGpCXQZzIMX71qIXrJi0plpvhTb77DBMnZBZOMospQ+7Ssh8BAAAA
14, KKZnIR0wa7JnCOEOe+0MbPLrMNc8I9Hu22fDD/iU5tIiYS+NwnwCAAAA
15, KK5afuhiG4JzEMGtV0Id4oe7hZKta46imNd6UHynVzhskFsDYqkAAAAA
16, KHvST2E7CskzVqg8ZL0DuWZyRGkJrEBTN8Qif2xhrLvCifowKO8BAAAB
17, KM7BRDKfDn7HJQZ7vtCMLvL9oBSr5rAGM0ObeHMJUQySWwY4Tw0DAAAA
18, KNJqKTfF6H2/Z0jqKa947fdjOj0q7GkfnCW5OI6Gf6/cmdAk0b0BAAAB
19, KARZ1j5kQOQ5pV1xSZD8vIBCda56BYXShiUeGe1z7Vdm72CxjagAAAAB
20, KMiENTJvXJpJYrWiooX98HhNzLuvgC0iisFjL0gNK8JBkETCrRMAAAAB
```

**Comment by psteckler** [comment]
Can the file be renamed to `0014-`... ? My pending RFC for module versioning is using 0013.

File: rfcs/0013-address-encoding.md (hunk 1 above)
Comment (psteckler) [reply rfcs/0013-address-encoding.md]: We could invert their capitalization. :-)

Comment (o1brad) [reply rfcs/0013-address-encoding.md]: Good point, @enolan. I agree we should have a distinctive prefix that's human readable, and have incorporated that suggestion into the RFC.

**Comment by cmr** [comment]
General note: the least-conflict thing to do with rfc numbers is to use `0000-rfc-name.md` and then number it immediately before merging.

**Comment by cmr** [comment]
@jkrauska the current public key encoding is a base64 of the bin_prot serialization of a record. There's some uniformity in how the serialization works.

**Comment by o1brad** [comment]
@es92 @imeckler can you review?

File: rfcs/0014-address-encoding.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+I recommend we follow the Bitcoin design, as laid out [here](https://en.bitcoin.it/wiki/Base58Check_encoding#Base58_symbol_chart).
+
+We should also add a unique prefix to prevent Coda from being accidentally sent to Bitcoin addresses or vice versa.
```
Comment (bkase) [inline rfcs/0014-address-encoding.md]: can you specify this prefix here that we'll use in the parity-check hashing step as @enolan proposed (just pick an arbitrary string -- "coda address" as he suggested is probably good)

Comment (o1brad) [reply rfcs/0014-address-encoding.md]: done

**Pull request merged**